### PR TITLE
Replacing prompt with popup dialog box for creating new circuit and saving project online

### LINKED ISSUE #1133: Replacing prompt with popup dialog box for new circuit
**Is your feature request related to a problem? Please describe.**
Its annoying to look at the prompt that comes up to create a new circuit
**Describe the solution you'd like**
Replacing prompt with a popup dialog box for creating new circuit like inserting new subcircuit

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by darkshredder**
Fixes #1133 

#### Describe the changes you have made in this PR -
I have Removed the prompt and added the script to create a new dialog box for creating new circuit and saving project online
#### New Circuit
ezgif com-video-to-gif (14)
####  saving project online
ezgif com-video-to-gif (16)

### CODE REVIEW

**Comment by darkshredder** [comment]
@tachyons @aayush-05 @nitin10s please review it and tell me how to fix complexity issue too

**Comment by darkshredder** [comment]
@Shivansh2407 try it out now it will work i guess

**Comment by darkshredder** [comment]
@Shivansh2407 fixed both issues check now
@tachyons @nitin10s @aayush-05 @satu0king please review it.

**Comment by darkshredder** [comment]
@Shivansh2407 I m working for all prompts, will create a PR soon

**Comment by Shivansh2407** [comment]
Ok cool, I think you can fix all the prompts in this PR only. Change the PR Title when you do so 👍🏻

**Comment by darkshredder** [comment]
@tachyons @satu0king @nitin10s @aayush-05 please review it

**Comment by darkshredder** [comment]
@Shivansh2407 I have made X working instead of creating cancel button as it looks weird to make a "X"
and cancel button

**Comment by darkshredder** [comment]
@Shivansh2407 I have seen that there are lot of prompt in circuit elements, will do it in another PR, as it will take time to get 7-8 prompt to converted to dialog box

**Comment by darkshredder** [comment]
@tachyons @satu0king @nitin10s @aayush-05 please review it

**Comment by darkshredder** [comment]
@tachyons @satu0king @nitin10s @aayush-05 please review it

File: app/views/simulator/edit.html.erb [hunk 1]
```diff
@@ -214,6 +214,8 @@
       <div id="combinationalAnalysis" style="display:none;" title="BooleanLogicTable"></div>
       <div id="insertSubcircuitDialog" style="display:none;" title="Insert SubCircuit"></div>
       <div id="openProjectDialog" style="display:none;" title="Open Project"></div>
+      <div id="saveonline" style="display:none;" title="Project Name"></div>
+      <div id="newcircuitcreate" style="display:none;" title="Create New Circuit"></div>
```
Comment (satu0king) [inline app/views/simulator/edit.html.erb]: I know the initial design had separate div for each dialog, but is it possible to have a common div for each?

File: public/js/data.js [hunk 2]
```diff
@@ -354,95 +384,12 @@ function deleteCurrentCircuit() {
         showMessage("Circuit was not deleted")
 }
 
-// Generates JSON of the entire project
-function generateSaveData(name) {
```
Comment (satu0king) [inline public/js/data.js]: Why are there so many changes here? Is it just movement of code or have you made actual changes?

**Comment by darkshredder** [comment]
@satu0king movement of code, Done because prompt blocks the line of code but Dialog is Synchronous in nature

File: app/views/simulator/edit.html.erb (hunk 1 above)
Comment (darkshredder) [reply app/views/simulator/edit.html.erb]: I guess there can be 1 div inside that these div's be present

Comment (darkshredder) [reply app/views/simulator/edit.html.erb]: or like circuit elements we can create div

**Comment by darkshredder** [comment]
@satu0king please review it again.

**Pull request closed without merging**